waspc/data/Generator/templates/sdk/wasp/client/webSocket/WebSocketProvider.tsx

```diff
@@ -36,7 +36,7 @@ apiEventsEmitter.on('sessionId.set', refreshAuthToken)
 apiEventsEmitter.on('sessionId.clear', refreshAuthToken)
 
 // PRIVATE API
-export const WebSocketContext: Context<WebSocketContextValue> = createContext({
+export const WebSocketContext: Context<WebSocketContextValue> = createContext<WebSocketContextValue>({
```

Comment:
Without explicitly setting the type argument to `WebSocketContextValue`, the function `createContext` infers it as the narrowest possible type of the passed object literal.

The inferred type of `createContext`'s return value is therefore:
```typescript
Context<{ 
  socket: Socket<ServerToClientEvents, ClientToServerEvents>, 
  isConnected: false
}>
```

Since `Context<T>` is contravariant for `T`<sup>1</sup>, the compiler complains when we try to assign `Context<{ socket: Socket<...>, isConnected: false}>` to `Context<{ socket: Socket<...>, isConnected: boolean}>`.

[1] This is thanks to a function call over `T` deep inside the `Context`. See https://github.com/wasp-lang/wasp/pull/989#discussion_r1101155001 for more on variance.